qa/rpc-tests/bip9-softforks.py

```diff
@@ -0,0 +1,220 @@
+            block.rehash()
+            block.solve()
+            test_blocks.append([block, True])
+            self.last_block_time += 1
+            self.tip = block.sha256
+            self.height += 1
+        return test_blocks
+
+    def get_bip9_status(self, key):
+        info = self.nodes[0].getblockchaininfo()
+        for row in info['bip9_softforks']:
+            if row['id'] == key:
+                return row
+        raise IndexError ('key:"%s" not found' % key)
+
+
+    def test_BIP(self, bipName, activated_version, invalidate, invalidatePostSignature):
+        # generate some coins for later
+        self.coinbase_blocks = self.nodes[0].generate(2)
+        self.height = 3  # height of the next block to build
+        self.tip = int ("0x" + self.nodes[0].getbestblockhash() + "L", 0)
+        self.nodeaddress = self.nodes[0].getnewaddress()
+        self.last_block_time = time.time()
+
+        assert_equal(self.get_bip9_status(bipName)['status'], 'defined')
+
+        # Test 1
+        # Advance from DEFINED to STARTED
+        test_blocks = self.generate_blocks(141, 4)
+        yield TestInstance(test_blocks, sync_every_block=False)
+
+        assert_equal(self.get_bip9_status(bipName)['status'], 'started')
+
+        # Test 2
+        # Fail to achieve LOCKED_IN 100 out of 144 signal bit 1
+        # using a variety of bits to simulate multiple parallel softforks
+        test_blocks = self.generate_blocks(50, activated_version) # 0x20000001 (signalling ready)
+        test_blocks = self.generate_blocks(20, 4, test_blocks) # 0x00000004 (signalling not)
+        test_blocks = self.generate_blocks(50, activated_version, test_blocks) # 0x20000101 (signalling ready)
+        test_blocks = self.generate_blocks(24, 4, test_blocks) # 0x20010000 (signalling not)
+        yield TestInstance(test_blocks, sync_every_block=False)
+
+        assert_equal(self.get_bip9_status(bipName)['status'], 'started')
+
+        # Test 3
+        # 108 out of 144 signal bit 1 to achieve LOCKED_IN
+        # using a variety of bits to simulate multiple parallel softforks
+        test_blocks = self.generate_blocks(58, activated_version) # 0x20000001 (signalling ready)
+        test_blocks = self.generate_blocks(26, 4, test_blocks) # 0x00000004 (signalling not)
+        test_blocks = self.generate_blocks(50, activated_version, test_blocks) # 0x20000101 (signalling ready)
+        test_blocks = self.generate_blocks(10, 4, test_blocks) # 0x20010000 (signalling not)
+        yield TestInstance(test_blocks, sync_every_block=False)
+
+        assert_equal(self.get_bip9_status(bipName)['status'], 'locked_in')
+
+        # Test 4
+        # 143 more version 536870913 blocks (waiting period-1)
+        test_blocks = self.generate_blocks(143, 4)
+        yield TestInstance(test_blocks, sync_every_block=False)
+
+        assert_equal(self.get_bip9_status(bipName)['status'], 'locked_in')
+
+        # Test 5
+        # Check that the new rule is enforced
```

Comment:
nit: I think this comment should say `# Check that the new rule is not enforced` ?